### PR TITLE
Address some "unused code" errors

### PR DESCRIPTION
**Pull request by michaelkirk**
# Issue

> What issue is this PR targeting? If there is no issue that addresses the problem, please open a corresponding issue and link it here.

I'm assuming a tracking issue is overly ceremonious in this case, but let me know.

## Tasklist

None of these seemed applicable, but let me know if I'm off base:

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

### CODE REVIEW

File: src/thor/astar_bss.cc [hunk 1]
```diff
@@ -513,7 +513,9 @@ std::vector<PathInfo> AStarBSSAlgorithm::FormPath(baldr::GraphReader& graphreade
   // Work backwards from the destination
   std::vector<PathInfo> path;
   travel_mode_t old = travel_mode_t::kPedestrian;
+  #ifndef NDEBUG
   int mode_change_count = 0;
```
Comment (kevinkreiser) [inline src/thor/astar_bss.cc:517]: you can use `[[maybe_unused]]` instead

Comment (michaelkirk) [reply src/thor/astar_bss.cc]: done.

File: test/logging.cc [hunk 2]
```diff
@@ -57,12 +57,12 @@ TEST(Logging, FileLoggerTest) {
     results.emplace_back(std::async(std::launch::async, work));
   }
 
-  // dont really care about the results but we can pretend
-  int exit_code = 0;
   for (auto& result : results) {
     try {
       result.get();
```
Comment (kevinkreiser) [inline test/logging.cc:62]: instead of doing try catch at all could you just do this:

```suggestion
      ASSERT_NO_THROW(result.get());
```

Comment (michaelkirk) [reply test/logging.cc]: done.

**Comment by kevinkreiser** [comment]
its a shame ci cant run properly with your prs

**Pull request merged**